drivers/docker/driver.go

```diff
@@ -616,17 +616,23 @@ func (d *Driver) createImage(task *drivers.TaskConfig, driverConfig *TaskConfig,
 	}
 
 	// Download the image
-	return d.pullImage(task, driverConfig, client, repo, tag)
+	imageID, imageUser, err := d.pullImage(task, driverConfig, repo, tag)
+
+	// validate the image user (windows only)
+	if err := d.validateImageUser(imageUser, task.User, driverConfig); err != nil {
+		return "", err
+	}
```

Comment:
Suppose we pull the image and find that it's running as `"ContainerAdmin"`. If I try to run the task again on the same node, the image may not have been GC'd. In that case, we'll hit the case above:

> // We're going to check whether the image is already downloaded.

And then skip this validation entirely! I think we want to either copy this method into the case above, or move it into the caller so that any image will get checked.

Reply:
good catch. We validate in the caller now: 492a7501c1a2e53ac798cefc3990ac0eba4af492